web/src/topic_list.ts

```diff
@@ -259,19 +274,20 @@ export function rebuild($stream_li, stream_id) {
 
 // For zooming, we only do topic-list stuff here...let stream_list
 // handle hiding/showing the non-narrowed streams
-export function zoom_in() {
+export function zoom_in(): void {
     zoomed = true;
 
     const stream_id = active_stream_id();
-    if (!stream_id) {
+    if (stream_id === undefined) {
         blueslip.error("Cannot find widget for topic history zooming.");
         return;
     }
 
     const active_widget = active_widgets.get(stream_id);
+    assert(active_widget !== undefined);
 
-    function on_success() {
-        if (!active_widgets.has(stream_id)) {
+    function on_success(): void {
+        if (!active_widgets.has(stream_id!)) {
```

Comment:
using the `!` here and below, because it doesn't understand we already checked they were undefined outside the scope of the function